File: leetcode/src/main/golang/is_palindrome.go
func isPalindrome(x int) bool {
    if x < 0 {
        return false
    }
    if x >= 0 && x < 10 {
        return true
    }
    if x % 10 == 0 {
        return false
    }
    rev := 0
    for ; rev < x; x /= 10 {
        rev = rev * 10 + x % 10
    }
    return x == rev || x == rev / 10
}

File: leetcode/src/main/ruby/is_palindrome.rb
# @param {Integer} x
# @return {Boolean}
def is_palindrome(x)
    return false if x < 0
    return true if x >= 0 && x < 10
    return false if x % 10 == 0

    rev = 0
    while rev < x
        rev = 10 * rev + x % 10
        x /= 10
    end
    return x == rev || x == rev / 10
end